quantus/metrics/faithfulness/monotonicity.py

```diff
@@ -371,7 +311,54 @@ def evaluate_batch(
         scores_batch:
             The evaluation results.
         """
-        return [
-            self.evaluate_instance(model=model, x=x, y=y, a=a)
-            for x, y, a in zip(x_batch, y_batch, a_batch)
-        ]
+        # Prepare shapes. Expand a_batch if not the same shape
+        if x_batch.shape != a_batch.shape:
+            a_batch = np.broadcast_to(a_batch, x_batch.shape)
+
+        batch_size = a_batch.shape[0]
+        a_batch = a_batch.reshape(batch_size, -1)
+        n_features = a_batch.shape[-1]
+
+        # Get indices of sorted attributions (ascending).
+        a_indices = np.argsort(a_batch, axis=1)
+
+        n_perturbations = math.ceil(n_features / self.features_in_step)
+        preds = []
+
+        # Copy the input x but fill with baseline values.
+        baseline_value = utils.get_baseline_value(
+            value=self.perturb_func.keywords["perturb_baseline"],  # type: ignore
+            arr=x_batch.reshape(batch_size, -1),
+            return_shape=(
+                batch_size,
+                n_features,
+            ),  # TODO. Double-check this over using = (1,).
```

Comment:
this `TODO` would need a bit more detail

Reply:
This is a relic of the past implementation, accidentally left it in the new one as well. I have deleted the TODO in the latest commit.